### ISSUE TITLE
Problems with report generation

### ISSUE THREAD

**Issue by rbedkowski** [kriskbx/gitlab-time-tracker#55]
Hi,

I have an issue with generation of a monthly report that includes activities from all GitLab projects:

```
 /usr/local/bin/gtt report time-tracker --type=group --from="2018-05-01" --to="2018-05-31" --verbose
🔍  Selected projects: ***
🔍  Resolving "time-tracker"... ✓
📦  Fetching issues... ✓
📦  Fetching merge requests... ✓
📎  Merging reports... ✓
⚙️️  Processing issues... ✗245) [===-------------------------------------] 7% - <1m left
Error: could not process issues.
RangeError: Invalid array length
    at Number.padLeft (/usr/local/share/.config/yarn/global/node_modules/gitlab-time-tracker/src/models/time.js:13:12)
    at Function.toHumanReadable (/usr/local/share/.config/yarn/global/node_modules/gitlab-time-tracker/src/models/time.js:114:37)
    at resolve (/usr/local/share/.config/yarn/global/node_modules/gitlab-time-tracker/src/models/hasTimes.js:130:41)
    at new Promise (<anonymous>)
    at promise.then (/usr/local/share/.config/yarn/global/node_modules/gitlab-time-tracker/src/models/hasTimes.js:110:38)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:182:7)
```

Can you please advice?

Regards,
Radek

**Comment by kriskbx** [comment]
This seems to be an issue with your specific set of data. Can you please send me a dump via mail so I can debug it? Just run the command again with ` --output=dump --file=/path/to/somwhere.json` and check the resulting json for sensitive information prior to sending it (secret tokens are deleted from the dump by default). [Check this out](https://github.com/kriskbx/gitlab-time-tracker#dumps) for more information on dumps.

**Comment by rbedkowski** [comment]
Hi,
When I execute the command with --output and --file parameters it finishes with the same error but does not create a dump file.

**Comment by kriskbx** [comment]
You're totally right. I reworked the dump mechanic in version 1.7.8. Can you upgrade and try again please?

**Comment by rbedkowski** [comment]
Hi Kris,

Thank you for this change. I've sent you the dump I've generated. As mentioned in the email, problem seems to be related to "body" not being an array in one case:

```
"body": {
	"id": 84,
	"description": "",
	"name": "admin",
...
```
instead of for example:
```
"body": [{
	"id": 279,
	"iid": 2,
	"project_id": 84,
...
```

Regards,
Radek

**Comment by kriskbx** [comment]
Thanks to your data I was able to nail the problem down: The time parsing is not working properly. You're using "/estimate 1mo" which is [not documented](https://docs.gitlab.com/ee/workflow/time_tracking.html#configuration) and not implemented on gtt's side (funny it works though).
I changed the regex and released a new version 1.7.10 which should fix this problem.

FYI: using months in GitLab time tracking is kinda buggy for some reason. If you "/spend 1mo" it results in "1mo 2w".
